test-netpol-enforcement/templates/unprotected-webserver.yaml

```diff
@@ -0,0 +1,83 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: unprotected-webserver
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: unprotected-webserver
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: unprotected-webserver
+    spec:
+      containers:
+        - name: nginx
+          image: "nginx:{{ .Chart.AppVersion }}"
+          ports:
+            - name: http
+              containerPort: 80
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: unprotected-webserver
+spec:
+  type: ClusterIP
+  selector:
+    app.kubernetes.io/name: unprotected-webserver
+  ports:
+    - port: 80
+      targetPort: http
+
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: unprotected-webserver81
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: unprotected-webserver81
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: unprotected-webserver81
+    spec:
+      containers:
+        - name: nginx
+          image: "nginx:{{ .Chart.AppVersion }}"
+          command:
+            - sh
+            - -c
+            - >-
+              sed -i.bak -r 's/listen\s+80;/listen 81;/' /etc/nginx/conf.d/default.conf &&
+              nginx -g 'daemon off;'
+          ports:
+            - name: http
+              containerPort: 81
+          readinessProbe:
+            httpGet:
+              path: /
+              port: http
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: unprotected-webserver81
+spec:
+  type: ClusterIP
+  selector:
+    app.kubernetes.io/name: unprotected-webserver81
+  ports:
+    - port: 81
```

Comment:
This port could be 80 as well i think, because i think what matters is the container port in the end for the network policy. Is that aligning with your understanding too?

Reply:
I don't know, you might be right!